ui/apps/platform/src/Containers/ComplianceEnhanced/Schedules/components/ReportJobs.tsx

```diff
@@ -58,22 +60,32 @@ type ReportJobsProps = {
 
 function ReportJobs({ scanConfig }: ReportJobsProps) {
     const { page, perPage, setPage, setPerPage } = useURLPagination(10);
-    const [filteredStatuses, setFilteredStatuses] = useState<RunState[]>([]);
-    const [showOnlyMyJobs, setShowOnlyMyJobs] = React.useState<boolean>(false);
+    const { searchFilter, setSearchFilter } = useURLSearch();
 
-    const onReportStatusFilterChange = (_checked: boolean, selectedStatus: RunState) => {
-        setFilteredStatuses((prevFilteredStatuses) => {
-            const isStatusIncluded = prevFilteredStatuses.includes(selectedStatus);
-            if (isStatusIncluded) {
-                return prevFilteredStatuses.filter((status) => status !== selectedStatus);
-            }
-            return [...prevFilteredStatuses, selectedStatus];
-        });
+    const filteredReportStates = ensureStringArray(searchFilter['Report state']);
+    const viewOnlyMyJobs = ensureBoolean(searchFilter['View only my jobs']);
+
+    const onReportStatesFilterChange = (_checked: boolean, selectedStatus: RunState) => {
+        const isStatusIncluded = filteredReportStates.includes(selectedStatus);
+        if (isStatusIncluded) {
+            setSearchFilter({
+                ...searchFilter,
+                'Report state': filteredReportStates.filter((status) => status !== selectedStatus),
+            });
+        } else {
+            setSearchFilter({
+                ...searchFilter,
+                'Report state': [...filteredReportStates, selectedStatus],
+            });
+        }
         setPage(1);
     };
 
     const onMyJobsFilterChange = (checked: boolean) => {
-        setShowOnlyMyJobs(checked);
+        setSearchFilter({
+            ...searchFilter,
+            'View only my jobs': String(checked),
```

Comment:
After having earlier this week discussed a boolean/string property, I am curious if there is already, or needs to be, a reverse conversion somewhere, for example:
https://github.com/stackrox/stackrox/pull/12725/files#diff-9a87fe2fecd74d0d7c4e54d98f08f9fe8f76480e1b90b1efbffadff19ede83e5R33

Reply:
We have an `ensureBoolean` used when accessing the search filter "value", if that's what you meant.